src/result/result.py

```diff
@@ -232,7 +232,16 @@ def unwrap(self) -> NoReturn:
         """
         Raises an `UnwrapError`.
         """
-        raise UnwrapError(self, "Called `Result.unwrap()` on an `Err` value")
+        if isinstance(self.value, Exception):
+            raise UnwrapError(
+                self,
+                f"Called `Result.unwrap()` on an `Err` value: {self.value.__repr__()}",
+            ) from self.value
```

Comment:
this needs test coverage. `exc.__cause__` is what it should look at

Reply:
Yeah it's very much a work in progress, I'll remove the draft label when it's ready for review